core/src/main/scala/io/qbeast/core/model/PreCommitHook.scala

```diff
@@ -46,7 +45,7 @@ trait PreCommitHook {
    * @return
    *   The output of the hook as a `PreCommitHookOutput`.
    */
-  def run(actions: Seq[Action]): PreCommitHookOutput
+  def run(actions: Seq[QbeastFile]): PreCommitHookOutput
```

Comment:
The purpose of the `PreCommitHook` should be to inject any block of code to any action available to the table. It is true that we only manage write actions, so it's fair to assume the input would be a list of `QbeastFile`. But in the future, I would expect this to be extended to different operations (change schema, update, delete...)

If it's not for this PR, maybe open an issue and discuss about it. Wdyt @Jiaweihu08 ?

Reply:
There's indeed a narrowing of scope from delta `Action` to `QbeastFile.`
We need an issue to discuss this and expand the classes in qbeast for other available actions.

Reply:
Would open one 👍